### PR TITLE
chore: move item.js code to erpnext from bloomstack_core 

### PR DESCRIPTION
**Pull request by meshramsaroj**
**Filename:** bloomstack_core/bloomstack_core/public/js/item.js

**Deleted file PR:** https://github.com/Bloomstack/bloomstack_core/pull/745

**Dependent PR:** https://github.com/Bloomstack/erpnext/pull/1686

### CODE REVIEW

**Comment by sahil28297** [comment]
@meshramsaroj add proper PR title

File: erpnext/stock/doctype/item/item.js [hunk 1]
```diff
@@ -29,9 +29,44 @@ frappe.ui.form.on("Item", {
 		if (frm.doc.is_fixed_asset) {
 			frm.trigger("set_asset_naming_series");
 		}
+
+		frm.set_query("metrc_item_category", () => {
+			if (frm.doc.metrc_uom) {
+				return {
+					query: "erpnext.hook_events.item.metrc_item_category_query",
```
Comment (sahil28297) [inline erpnext/stock/doctype/item/item.js]: The path is not proper, coordinate with Myuddin for putting the correct path

File: erpnext/stock/doctype/item/item.js [hunk 2]
```diff
@@ -29,9 +29,44 @@ frappe.ui.form.on("Item", {
...
+						metrc_uom: frm.doc.metrc_uom
+					}
+				};
+			}
+		});
+
+		frm.set_query("metrc_uom", () => {
+			if (frm.doc.metrc_item_category) {
+				return {
+					query: "erpnext.hook_events.item.metrc_uom_query",
```
Comment (sahil28297) [inline erpnext/stock/doctype/item/item.js]: The path is not proper, coordinate with Myuddin for putting the correct path

File: erpnext/stock/doctype/item/item.js [hunk 3]
```diff
@@ -29,9 +29,44 @@ frappe.ui.form.on("Item", {
...
+						metrc_item_category: frm.doc.metrc_item_category
+					}
+				};
+			}
+		});
+
+		frm.set_query("metrc_unit_uom", () => {
+			if (frm.doc.metrc_item_category) {
+				return {
+					query: "erpnext.hook_events.item.metrc_unit_uom_query",
```
Comment (sahil28297) [inline erpnext/stock/doctype/item/item.js]: The path is not proper, coordinate with Myuddin for putting the correct path

File: erpnext/stock/doctype/item/item.js [hunk 4]
```diff
@@ -132,6 +167,52 @@ frappe.ui.form.on("Item", {
...
+	brand: (frm) => {
+		if (frm.is_new()) {
+			frm.trigger("build_item_code");
+		}
+	},
+
+	build_item_code: (frm) => {
+		// TODO: allow toggling autoname from the website
+		frappe.call({
+			method: "erpnext.hook_events.item.autoname_item",
```
Comment (sahil28297) [inline erpnext/stock/doctype/item/item.js]: The path is not proper, coordinate with Myuddin for putting the correct path

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Comment by sahil28297** [comment]
@meshramsaroj any update on this?

**Comment by meshramsaroj** [comment]
> @meshramsaroj any update on this?

Yes, I have added the correct path.
Please check

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**